### ISSUE TITLE
Support fetching node details through node label instead of ID

### ISSUE THREAD

**Issue by fbarl** [weaveworks/scope#3122]
If we could call `/api/topology/{name}/{id}` with node labels instead of IDs, we would be able to load node details directly if only node labels were provided in the URL.

As far as I can see, this is the only backend blocker for resolving #2030 and it would only require us to do a bit more sophisticated search through nodes in `handleNode` in `api_topology.go`.

**Comment by rade** [comment]
What exactly do you mean by "node labels"? The `label` field in the json? For k8s that does contain the k8s name, but don't we also need the namespace? Note that [the `rank` field combines namespace and name](https://github.com/weaveworks/scope/blob/v1.8.0/render/detailed/summary.go#L294).

**Comment by fbarl** [comment]
> What exactly do you mean by "node labels"? The `label` field in the json? 

Yeah, I did mean the `label` field originally, but it makes more sense to include namespace in the name. I guess `rank` is exactly what we want since for controllers it has the same format as our Deploy/Monitor workload IDs, it's just the name _rank_ that I find confusing :P
